cleanlab/experimental/tf_nlp.py

```diff
@@ -0,0 +1,74 @@
+"""
+A Scikeras classifier (adapted for huggingface models) which
+can be used for finding label issues in any nlp dataset.
+"""
+
+from scikeras.wrappers import KerasClassifier
+from sklearn.preprocessing import FunctionTransformer
+import tensorflow as tf
+import numpy as np
+
+
+class MultiInputKerasClassifier(KerasClassifier):
+    def __init__(self, train_input, seq_len, valid_input=None, y_val=None, **kwargs):
+        """
+        Basic scikeras/scikeras does not directly provide the option of having
+        a multi-input model. However, you can work around this problem by
+        adding the `feature_encoder` property to the class which extends
+        Scikitlearn's BaseEstimator.
+
+        References:
+        - https://towardsdatascience.com/scikeras-tutorial-a-multi-input-multi-output-wrapper-for-capsnet-hyperparameter-tuning-with-keras-3127690f7f28
+        - https://www.adriangb.com/scikeras/stable/notebooks/DataTransformers.html#4.-Multiple-inputs
+
+        """
+        super().__init__(**kwargs)
+        self.train_input = train_input
+        self.valid_input = valid_input
+        self.y_val = y_val
+        self.seq_len = seq_len
+
+    def split_input(self, X):
+        splitted_X = [
+            X[:, : self.seq_len],  # input_ids
+            X[:, self.seq_len :],  # attention_mask
+        ]
+        return splitted_X
+
+    @property
+    def feature_encoder(self):
+        return FunctionTransformer(
+            func=self.split_input,
+        )
+
+    def _get_tf_input(self, ids, train_input):
+        indexed_input_ids = np.array(tf.gather(train_input["input_ids"], indices=ids))
+
+        indexed_attention_mask = np.array(tf.gather(train_input["attention_mask"], indices=ids))
```

Comment:
This class only seems to work with very specifically formatted `train_input`, so needs to be documented in detail what exactly this format is and what it represents.

Reply:
I just noticed that when we will use non-strictly arrays most probably `train_input` won't have a particular format. So, if you agree then I would keep this part temporarily on hold.

Reply:
Right understood that `train_input` can have whatever format you like here.
I'm just suggesting that you can add a little documentation to describe what this particular `train_input` is supposed to look like for your specific class.  For example, `train_input` clearly has to have fields like: "input_ids", "attention_mask", etc, so for each of these fields:  can you just add a 1 sentence description of the field or provide a link to another website where the field is defined?

Thanks! Having some (even minimal) documentation will be very helpful to our users!